### DISCUSSION TITLE
telegraf@future: NEW MAINTAINER NEEDED

### DISCUSSION THREAD

**Discussion by wojpawlik** [telegraf/telegraf#1526] [General]
@dotcypress is inactive. @KnorpelSenf moved on to develop https://grammy.dev/. I, the last maintainer, am about to step down.

I already gave up trying to develop Telegraf 3 long ago. Yesterday I gave up on Telegraf 4 as well.

I tried everything to solve https://github.com/telegraf/telegraf/issues/1388 in Telegraf 4. And I failed. https://grammy.dev/ already uses the only approach that works.

Telegraf 4 was an important stepping stone to develop https://grammy.dev/. But I believe it no longer has a future.

Please write a comment below if you'd like to maintain Telegraf 3 or Telegraf 4.

If nobody steps up, I'll revoke my automation token ~~and archive the repository~~ after I release support for upcoming Bot API 5.4.

**Comment by KnorpelSenf** [comment]
Will update as needed, post questions in https://t.me/TelegrafJSChat or underneath

# FAQ

## What do I need in order to become the next maintainer?

Motivation to use your free time. It is good if you can picture yourself maintaining this also one year into the future.

## Okay but do I need to be a coding expert?

You need to know how to use Google to solve programming problems. Because that's what all good (JavaScript) programmers do. You don't need to be a coding expert, because Google will teach you what you need when you need it.

It helps to have experience with developing bots, and knowledge about the Bot API.

## Do I need to know TypeScript?

No. Just maintain Telegraf 3 and you're good to go.

## Is Telegraf going to be dead by the end of the week?

No.

**Reply by zeusricote** [reply]
Give It to me i Will maintain It.

**Reply by KnorpelSenf** [reply]
@zeusricote looking forward to seeing your contributions! You can coordinate your pull requests with the new maintainers

**Comment by MKRhere** [comment]
I'm interested; however the direction I prefer to go is try and fix the problems people have been having with Telegraf 4. PRs to v3 can still be verified and released as has been happening.

Also, I'm hoping to get nominal support from both of you @wojpawlik and @KnorpelSenf as I get used to the codebase. I've used Telegraf for fairly simple projects for over 3 years, and would like for the project to continue existing. More internal alignment with grammy would be good too. That's something I've been thinking of, as well.

**Reply by wojpawlik** [reply]
Great! @ me anytime you need help. I gave you publish rights and invited you to be an admin of the repo. You may want to replace `NPM_TOKEN` secret with your own automation token. I already revoked mine. https://github.com/telegraf/telegraf/blob/63541abcd9487802a525f2576b10a523ee3249fd/.github/workflows/release.yml#L3-L6

Feel free to invite someone else to help, I won't lock the discussion. And perhaps someone dedicated to Telegraf 3.

For context, we've worked together on several projects. Muthu is a trustworthy and capable successor.

**Reply by MKRhere** [reply]
Thanks! I've accepted and replaced `NPM_TOKEN`.

**Reply by KnorpelSenf** [reply]
I'm also available when I'm mentioned. I'll keep on maintaining `typegram`.

**Comment by dotcypress** [comment]
I'm inclined to continue working on v3.x branch and publish it as v5 in future.
I have a lot of peeps who wants to maintain 3.x. And probably I can help with that.

**Reply by wojpawlik** [reply]
Now I believe Telegraf 5 should be based on Telegraf 3. DO you agree @MKRhere?

Either way, I think it's too early for Telegraf 5. Node 16 adds (experimental) `Blob` and it'll be LTS soon.

**Reply by MKRhere** [reply]
So far I've only known documentation and types to be the major reported problems of v4. This seems to have created an unfair amount of backpressure towards v3. I intend to try and improve both in the coming weeks. I agree it's too early for v5 in either case, though.

**Reply by KnorpelSenf** [reply]
The whole discussion about publishing the old v3 as a v5 seems more like a statement that v4 gets reverted and v3 is the future. It doesn't seem to have any true advantages over v3 that would require a breaking chance/major version bump—except for undoing the breaking changes of v4. Is this accurate?

**Reply by wojpawlik** [reply]
This. What's your perspective @murka?

**Reply by murka** [reply]
I think so too, it's too early for the Telegraf 5. The v5 need more improves and fixes.

**Reply by wojpawlik** [reply]
Then I'll modify https://github.com/telegraf/telegraf/pull/1528 to be mergable as 3.40 as soon as my real life settles (up to few weeks)

**Reply by murka** [reply]
I think, we shouldn't be in such a hurry. Let be v5 as draft.

**Reply by wojpawlik** [reply]
Alright. I won't rush then.

Still, your changes are good. Why wait for 5.0? Why not release them as 3.40? And then work on 5.0.

**Comment by abdoh-alkhateeb** [comment]
I'm fairly new to the world of bots, but I do like the work and the effort that's been put into frameworks like the Telegraf. And even though, I'm not familiar enough with the codebase and the framework as a whole, I'm interested in maintaining it, I liked its concept, and I'm willing to delve deep into it.

However, I don't have any experience with Typescript, I'm a young full-stack developer, but I want to contribute to the community.

Would appreciate any guidance and tips from the previous maintainers.

**Comment by MDalprato** [comment]
I'm more than interest in maintaining.
I'm not a coding expert but I'm developing some react apps now and I know some skills with js.

**Comment by bokssssss** [comment]
hi hello i like this so much telegraf

**Comment by zeusricote** [comment]
perfect, have a good day people

Sent with [ProtonMail](https://protonmail.com/) Secure Email.

‐‐‐‐‐‐‐ Original Message ‐‐‐‐‐‐‐

**Comment by fzn0x** [comment]
how to and the rules to contact if you want to be maintainer?

**Comment by wojpawlik** [comment]
Please start by making pull requests y'all. It's a chance to get to know you, and explain the codebase.
